subsys/logging/log_core.c

```diff
@@ -644,6 +647,16 @@ char *log_strdup(const char *str)
 		return (char *)log_strdup_fail_msg;
 	}
 
+	if (IS_ENABLED(CONFIG_LOG_STRDUP_POOL_PROFILING)) {
+		size_t slen = strlen(str);
+		int key;
+
+		key = irq_lock();
+		log_strdup_max = MAX(++log_strdup_in_use, log_strdup_max);
```

Comment:
`MAX()` is defined as `#define MAX(a, b) (((a) > (b)) ? (a) : (b))` so doesn't this expand to:
```
((++log_strdup_in_use) > (log_strdup_max) ? (++log_strdup_in_use) : (log_strdup_max))
```

I.e. you'll end up incrementing log_strdup_in_use twice.

Reply:
thanks. fixed.